classes/observer.php

```diff
@@ -135,4 +137,334 @@ public static function mod_quiz_event_attempt_submitted(
+        $submissionid = $eventdata["other"]["submissionid"];
+
+        if (!plagiarism_copyleaks_moduleconfig::is_module_enabled('assign', $cmid)) {
+            return;
+        }
+
+        // Get user id.
+        $userid = $eventdata['relateduserid'];
+        if ($userid == null) {
+            $userid = $eventdata['userid'];
+        }
+
+        if ($eventdata['target'] == 'submission' && $eventdata['action'] == 'updated') {
+
+            $clfiles = $DB->get_records('plagiarism_copyleaks_files', ['cm' => $cmid, 'itemid' => $submissionid, 'submissiontype' => 'file']);
+            $fs = get_file_storage();
+            foreach ($clfiles as $clfile) {
+                $file = $fs->get_file_by_hash($clfile->identifier);
+                if ($file === false) {
+                    $cl = new \plagiarism_copyleaks_comms();
+                    $data = (array)[
+                        'identifier' => $clfile->identifier,
+                        'courseModuleId' => $clfile->cm,
+                        'moodleUserId' => $clfile->userid,
+                    ];
+                    $cl->delete_report($data);
+                    // $DB->delete_records('plagiarism_copyleaks_files', ['cm' => $cmid, 'userid' => $clfile->userid, 'identifier' => $clfile->identifier]);
+                }
+            }
+        }
+    }
+
+    /**
+     * assign submission comment creation event handler.
+     * @param \assignsubmission_comments\event\comment_created $event
+     */
+    public static function assign_submission_comment_created(
+        \assignsubmission_comments\event\comment_created $event
+    ) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+
+        global $DB;
+        $cl = new \plagiarism_copyleaks_comms();
+        $datetime = new DateTime();
+        $eventdata = $event->get_data();
+        $commentcontent = $DB->get_record('comments', ['id' => $eventdata['objectid']], 'content');
+        $commentdata = (array)[
+            'commentId' => $eventdata['objectid'],
+            'courseModuleId' => $eventdata['contextinstanceid'],
+            'moodleUserId' => $eventdata['userid'],
+            'content' => $commentcontent->content,
+            'createdAt' => ($datetime->setTimestamp($eventdata['timecreated']))->format('Y-m-d H:i:s'),
+        ];
+
+        $submissionid = $eventdata['other']['itemid'];
+
+        $cl->add_assign_submission_comment($commentdata, $submissionid);
+    }
+
+    /**
+     * assign submission comment deletion event handler.
+     * @param \assignsubmission_comments\event\comment_deleted $event
+     */
+    public static function assign_submission_comment_deleted(
+        \assignsubmission_comments\event\comment_deleted $event
+    ) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+
```

Comment:
@ShadeAma shouldn't we check if assign is enabled here? same as other events

if (!plagiarism_copyleaks_moduleconfig::is_module_enabled('assign', $cmid)) {
return;
}